react/components/EditorContainer/Sidebar/BlockEditor/BlockConfigurationList/utils.ts

```diff
@@ -3,6 +3,36 @@ import { ListContentData } from '../../../queries/ListContent'
 
 import { GetDeleteStoreUpdater } from './typings'
 
+export const isConfigurationExpired = (
+  configuration: ExtensionConfiguration
+) => {
+  const parsedDate = configuration.condition.statements.reduce(
+    (acc?: Date, curr?: ExtensionConfigurationConditionStatement) => {
+      if (acc) {
+        return acc
+      }
+      if (curr && curr.subject === 'date') {
+        const value = JSON.parse(curr.objectJSON)
+        if (value.to) {
+          return new Date(value.to)
+        }
+      }
+      return undefined
+    },
+    undefined
+  )
+
+  if (parsedDate && parsedDate < new Date()) {
+    return true
+  }
+
+  return false
+}
+
+export const isConfigurationScheduled = (
+  configuration: ExtensionConfiguration
+) => configuration.condition.statements.length > 0
```

Comment:
This will work for now, but once we have more condition types we'll also have to check if any of the `configuration.condition.statements` has a `'date'` subject.

Reply:
That is a relevant observation. I'd vote for doing it just in time, not just in case.